conans/test/unittests/client/generators/scons_test.py

```diff
@@ -33,3 +33,22 @@ def variables_setup_test(self):
         self.assertIn('    "conan_version" : "None",', scons_lines)
         self.assertIn('    "MyPkg_version" : "0.1",', scons_lines)
         self.assertIn('    "MyPkg2_version" : "3.2.3",', scons_lines)
+
+    def system_frameworks_libs_test(self):
+        # https://github.com/conan-io/conan/issues/7301
+        conanfile = ConanFile(TestBufferConanOutput(), None)
+        conanfile.initialize(Settings({}), EnvValues())
+        cpp_info = CppInfo("MyPkg", "")
+        cpp_info.version = "0.1"
+        cpp_info.libs = ["mypkg"]
+        cpp_info.system_libs = ["pthread"]
+        cpp_info.frameworks = ["cocoa"]
+        cpp_info.frameworkdirs = ["frameworks"]
+        cpp_info.filter_empty = False
+        conanfile.deps_cpp_info.add("MyPkg", cpp_info)
```

Comment:
After the components changes, the object `deps_cpp_info` aggregates `DepCppInfo` objects:

```suggestion
        conanfile.deps_cpp_info.add("MyPkg", DepCppInfo(cpp_info))
```